src/gui/layout/qgslayoutviewtooladdnodeitem.cpp

```diff
@@ -74,7 +74,15 @@ void QgsLayoutViewToolAddNodeItem::layoutPressEvent( QgsLayoutViewMouseEvent *ev
       return;
 
     if ( QgsLayoutNodesItem *nodesItem = qobject_cast< QgsLayoutNodesItem * >( item ) )
+    {
       nodesItem->setNodes( mPolygon );
+      if ( !nodesItem->isValid() )
+      {
+        nodesItem->deleteLater();
```

Comment:
why deleteLater here? I suggest using std::unique_ptr  so the node item gets destroyed automatically on return, and release it on addLayoutItem

Reply:
Ok